### PR TITLE
Added CinemaHallController, OrderController, ShoppingCartController, UserController with all necessary mappers and DTOs

### PR DESCRIPTION
**Pull request by KostiantynVorobiov**

### CODE REVIEW

**Review by DmGrt — COMMENTED** [review]
Seems everything is fine)

File: src/main/java/com/cinema/model/dto/UserResponseDto.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.cinema.model.dto;
+
+import lombok.Data;
+
+@Data
+public class UserResponseDto {
+    private Long id;
+    private String email;
+    private String password;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/UserResponseDto.java]: let's not return secure data

File: src/main/java/com/cinema/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -25,10 +24,10 @@ public AuthenticationServiceImpl(UserService userService,
 
     @Override
     public User login(String email, String password) throws AuthenticationException {
-        Optional<User> userFromDb = userService.findByEmail(email);
-        if (userFromDb.isPresent() && userFromDb.get().getPassword()
-                .equals(HashUtil.hashPassword(password, userFromDb.get().getSalt()))) {
-            return userFromDb.get();
+        User userFromDb = userService.findByEmail(email);
+        if (userFromDb != null && userFromDb.getPassword()
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/security/AuthenticationServiceImpl.java]: first - can it be null?

second - the problem is that earlier, when your user entered incorrect email, the AuthenticationException was thrown with the informative message

and now the NoSuchElementException is thrown

my advice is not to change `findByEmail()` method
but I think you won't like calling optional.get() all the time in your controllers, so let's create `getById(Long userId)` and pass userId as the `@RequestParam` to the controllers?
what do you think?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -58,4 +58,14 @@ public User add(User user) {
             throw new DataProcessingException("Can't find user by email " + email, e);
         }
     }
+
+    @Override
+    public Optional<User> get(Long id) {
+        logger.info("Trying to get user by id");
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.of(session.get(User.class, id));
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/dao/impl/UserDaoImpl.java]: what happens when we call Optional.of(null)?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**